docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+
+Figure 4. Class Diagram of the Model component
+
+API: Model.java
+
+The Model component,
+* Holds all the in-memory data of type event, quiz and contact.
+* Each data type has a corresponding manager that the controller can interface with. This is named as <datatype>Manager.java.
+* During a command execution, the manager will handle the modification and reading of its data type.
+* After the execution, the corresponding output is then passed on to the User Interface component to be shown to the user.
+
+<br />
+<br />
+
+#### Storage component
+![storagecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/StorageComponent.png)
+
+Figure 5. Class Diagram of the Storage component
+
+API: StorageManager.java
+
+The Storage component,
+* Creates the necessary data files for the operation of Plan&Score.
+* Reads encoded data from Plan&Score’s data files.
+* Writes encoded data to Plan&Score’s data files.
+
+There are 3 categories of data stored by Plan&Score: `event`, `quiz` and `config`
+
+Plan&Score loads data automatically from .txt files in the `data` directory.
+Each `StorageManager` reads in their respective data files through a `decoder` and writes to the same file through an `encoder`
+
+##### Event Storage
+![eventstorage](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/EventStorageManager.png)
+
+Figure 6. Class Diagram of EventStorageManager
+
+API: EventStorageManager.java
+`EventStorageManager` is responsible for the reading and writing of data from Plan&Score’s `events.txt` file, located in the `{root}/data` directory.
+
+It utilises a decoder (`EventListDecoder.java`) for the reading of data, and an encoder (`EventListEncoder.java`) for the writing of data. 
+
+###### Reading events
+
+`EventListDecoder` is responsible for the decoding of data from `events.txt`.
+
+It returns an `EventParameter` to `EventStorageManager`. 
+
+![eventreadstorage](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/EventStorageReadSequence.png)
+
+Figure 7. Sequence Diagram of the reading of data
+
+###### Writing events
+
+`EventListEncoder` is responsible for the creation of the encoded strings for `EventStorageManager` to write to `events.txt`.
+
+It returns a `String` to `EventStorageManager` for further writing. 
+
+![eventwritestorage](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/EventStorageSaveSequence.png)
+
+Figure 8. Sequence Diagram of the writing of data
+
+##### Quiz Storage
+API: QuizStorageManager.java
+
+![quizwritestorage](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/QuizWriteStorage.png)
+
+Figure 9. Sequence Diagram of the writing of data
+
+![quizreadstorage](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/QuizReadStorage.png)
```

Comment:
![image](https://user-images.githubusercontent.com/57870728/97398651-8a9e1680-1926-11eb-8492-cb7d6e8a8b14.png)

Should decodeQuizFromString be written as decodeQuizFromString()?